src/main/scala/com/snowflake/snowpark/internal/ParameterUtils.scala

```diff
@@ -106,36 +106,45 @@ private[snowpark] object ParameterUtils extends Logging {
   private[snowpark] def parsePrivateKey(key: String): PrivateKey = {
     try {
       val decoded = Base64.decodeBase64(key)
-      val derReader = new DerInputStream(decoded)
-      val seq = derReader.getSequence(0)
-
-      if (seq.length < 9) {
-        throw new GeneralSecurityException("Could not parse a PKCS1 private key.")
-      }
-
-      // seq(0) is version, skip
-      val modulus = seq(1).getBigInteger
-      val publicExp = seq(2).getBigInteger
-      val privateExp = seq(3).getBigInteger
-      val prime1 = seq(4).getBigInteger
-      val prime2 = seq(5).getBigInteger
-      val exp1 = seq(6).getBigInteger
-      val exp2 = seq(7).getBigInteger
-      val crtCoef = seq(8).getBigInteger
-      val keySpec = new RSAPrivateCrtKeySpec(
-        modulus,
-        publicExp,
-        privateExp,
-        prime1,
-        prime2,
-        exp1,
-        exp2,
-        crtCoef)
-      val keyFactory = KeyFactory.getInstance("RSA")
-      keyFactory.generatePrivate(keySpec)
+      val kf = KeyFactory.getInstance("RSA")
+      val keySpec = new PKCS8EncodedKeySpec(decoded)
+      kf.generatePrivate(keySpec)
     } catch {
       case e: Exception =>
-        throw ErrorMessage.MISC_INVALID_RSA_PRIVATE_KEY(e.getMessage)
```

Comment:
minor request: Could retain the PKCS#8 attempt's exception's detail in the catch/re-throw when a PKCS#1 attempt also subsequently fails? The original exception's information could be helpful when working with a valid PKCS#8 key that's facing errors when read in Java.

In the current approach we'd just receive a blanket `Could not parse a PKCS1 private key` when passing a PKCS#8 value that's not readable through Java (there are a few algorithms that JSSE [does not support](https://community.snowflake.com/s/article/Private-key-provided-is-invalid-or-not-supported-rsa-key-p8--data-isn-t-an-object-ID)).

Reply:
Sure, great suggestion. I changed it to print error messages of both `pkcs1` and `pkcs8`.